up42/auth.py

```diff
@@ -100,41 +123,55 @@ def __init__(
         except KeyError:
             self.authenticate = True
 
+        if all([arg in kwargs for arg in ["project_id", "project_api_key"]]) and auth_type == AuthType.PROJECT.value:
+            self.credentials_id = kwargs["project_id"]
+            self.credentials_key = kwargs["project_api_key"]
+            self.project_id = self.credentials_id
+
         if self.authenticate:
             self._find_credentials()
             self._get_token()
-            self._get_workspace()
+            self._get_user_id()
             logger.info("Authentication with UP42 successful!")
 
     def __repr__(self):
         env_string = f" ,{self.env}" if self.env != "com" else ""
-        return f"UP42ProjectAuth(project_id={self.project_id}{env_string})"
+        if self.auth_type == AuthType.ACCOUNT.value:
+            return f"UP42UserAuth(user_id={self.credentials_id}{env_string})"
+        return f"UP42ProjectAuth(project_id={self.credentials_id}{env_string})"
 
     def _find_credentials(self) -> None:
```

Comment:
We need to split this into 2 methods. One will choose the credentials source between config file and arguments, something like:
```python
    def _choose_credential_source(cfg_file: str, kwargs: dict):
        config = {}
        try:
            with open(cfg_file, encoding="utf-8") as src:
                config.update(json.load(src))
            logger.info("Got credentials from config file.")
        except FileNotFoundError as e:
            raise ValueError("Selected config file does not exist!") from e
        if set(config.keys()).intersection(set(kwargs.keys())):
            raise ValueError("Credentials must be provided either via config file or arguments, but not both")
        return config if config else kwargs
```
The second will read the parameters from the chosen source, smith like
```python
    def _set_credentials(self, source: dict):
        schemas = {
            "user": ("username", "password"),
            "project": ("project_id", "project_api_key")
        }
        token_retrievers = {"project": self._get_project_token, "user": self._get_account_token}

        for parameters, schema in schemas.items():
            if set(parameters).issubset(source.keys()):
                self._credentials_id = source[parameters[0]]
                self._credentials_key = source[parameters[1]]
                self._get_token = token_retrievers[schema]
        if self._credentials_id is None:
            # raise error
```

Reply:
The way it works currently for project-based authentication is: 

If there are both arguments and cfg_file, the cfg_file has the priority and we log that we choose the cfg_file over the parameters. 

My question is: 
1. Now we have 4 options. cfg_project_based, keys_project_based, cfg_account_based, keys_account_based. 
2. Should we follow that cfg_based are priority.
3. Also, I think that we should also allow project_based and account_based at the same time provided, but let the account based to have priority. 
4. The only case we raise an error should be with not complete credentials provided.

Reply:
For consideration, what about something like this to add some hierachy if the user adds project-based or account-based via cfg_files and arguments.
`
    def _choose_credentials(username, password, project_id, project_api_key, config_file) -> tuple[str, str]:
        if config_file:
            try:
                with open(config_file, 'r') as file:
                    config_data = json.load(file)
                    if 'project_id' in config_data and 'project_api_key' in config_data:
                        return config_data['project_id'], config_data['project_api_key'] 
                    if 'username' in config_data and 'password' in config_data:
                        return config_data['username'], config_data['password'] 
            except FileNotFoundError as e:
                raise ValueError("Selected config file does not exist!") from e

        if project_id and project_api_key:
            return project_id, project_api_key
        
        if username and password:
            return username, password 
        raise ValueError("No credentials provided either via config file or arguments.")

`